.github/workflows/cmake.yml

```diff
@@ -0,0 +1,32 @@
+name: CMake
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install apt package
+      run: sudo apt-get install -y doxygen
+
+    - name: Install pip packages
+      run: pip install -r third_party/requirements.txt
+
+    - name: Configure CMake
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+    - name: Generate source from spec
+      run: cmake --build ${{github.workspace}}/build --target check-generated --config ${{env.BUILD_TYPE}}
+
+    - name: Build
+      run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
```

Comment:
For me this --config param doesn't change anything. What is relevant here for setting up debug or release build is the -DCMAKE_BUILD_TYPE argument which you added in the previous step.

Reply:
I believe `--config` is important when using the Visual Studio project generator. This will be defaulting to the Unix Makefiles generation though.

Reply:
I used a template :) I'll remove that config part. Windows is probably going to need its own separate job anyway.

Reply:
Done